src/action/base/create_or_merge_nix_config.rs

```diff
@@ -0,0 +1,665 @@
+                        .settings_mut()
+                        .insert(pending_conf_name.to_owned(), pending_conf_value.to_owned());
+                }
+            }
+
+            if !unmergeable_config_names.is_empty() {
+                return Err(ActionError::Custom(Box::new(
+                    CreateOrMergeNixConfigError::UnmergeableConfig(
+                        unmergeable_config_names,
+                        this.path,
+                    ),
+                )));
+            }
+
+            if !merged_nix_config.settings().is_empty() {
+                this.nix_configs.merged_nix_config = merged_nix_config;
+                return Ok(StatefulAction::uncompleted(this));
+            }
+
+            tracing::debug!(
+                "Setting Nix configurations in `{}` already complete",
+                this.path.display()
+            );
+            return Ok(StatefulAction::completed(this));
+        } else {
+            let mut merged_nix_config = NixConfig::new();
+            for (pending_conf_name, pending_conf_value) in pending_nix_config.settings() {
+                merged_nix_config
+                    .settings_mut()
+                    .insert(pending_conf_name.to_owned(), pending_conf_value.to_owned());
+            }
+
+            if !merged_nix_config.settings().is_empty() {
+                this.nix_configs.merged_nix_config = merged_nix_config;
+            }
+        }
+
+        Ok(StatefulAction::uncompleted(this))
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "create_or_merge_nix_config")]
+impl Action for CreateOrMergeNixConfig {
+    fn tracing_synopsis(&self) -> String {
+        format!("Create or merge nix.conf file `{}`", self.path.display())
+    }
+
+    fn tracing_span(&self) -> Span {
+        let span = span!(
+            tracing::Level::DEBUG,
+            "create_or_merge_nix_config",
+            path = tracing::field::display(self.path.display()),
+            mode = tracing::field::display(format!("{:#o}", NIX_CONF_MODE)),
+            merged_nix_config = tracing::field::Empty,
+        );
+
+        if tracing::enabled!(tracing::Level::TRACE) {
+            span.record(
+                "merged_nix_config",
+                &self
+                    .nix_configs
+                    .merged_nix_config
+                    .settings()
+                    .iter()
+                    .map(|(k, v)| format!("{k}='{v}'"))
+                    .collect::<Vec<_>>()
+                    .join(" "),
+            );
+        }
+        span
+    }
+
+    fn execute_description(&self) -> Vec<ActionDescription> {
+        vec![ActionDescription::new(
+            self.tracing_synopsis(),
+            vec![format!(
+                "If {} already exists, we will attempt to merge the current settings with our settings; \
+                otherwise, it will be created with only our settings",
+                self.path.display()
+            )],
```

Comment:
Can this message mention which settings it will change? (I think we have them here anyways)